model/document.js

```diff
@@ -197,7 +197,32 @@ module.exports = memoize(function (db) {
 			if (statusLog.length) data.statusLog = statusLog;
 
 			return data;
-		} }
+		} },
+		toWSSchema: {
+			value: function (ignore) {
+				var schema = {
+					type: "array",
+					items: {
+						type: "object",
+						properties: {
+							code: {
+								type: "enum",
+								ref: "documents"
+							},
+							data: {}
+						}
+					}
+				};
+				schema.items.properties.files =
+					this.files.map._descriptorPrototype_.type.prototype.toWSSchema();
```

Comment:
Files are NestedMaps if I recall correctly. For cases of NestedMap instances I think it is cleaner to aquire, the type info through `getItemType` method